proto/info/edge_node_cluster.proto

```diff
@@ -0,0 +1,236 @@
+}
+
+enum KubePodStatus {
+  POD_STATUS_UNKNOWN = 0;
+  POD_STATUS_PENDING = 1;
+  POD_STATUS_RUNNING = 2;
+  POD_STATUS_SUCCEEDED = 3;
+  POD_STATUS_CONTAINER_CREATING = 4;
+  POD_STATUS_CRASHLOOP_BACKOFF = 5;
+  POD_STATUS_ERROR = 6;
+  POD_STATUS_EVICTED = 7;
+  POD_STATUS_FAILED = 8;
+}
+
+message KubeEVEAppPodInfo {
+  // Name of the EVE application
+  string name = 1;
+
+  // Application Status
+  KubePodStatus status = 2;
+
+  // Restart count of the application
+  uint32 restartCount = 3;
+
+  // Restart time of the application, seconds ago
+  uint64 restartTime = 4;
+
+  // Age of the application, seconds ago
+  uint64 age = 5;
+
+  // IP address of the application, on cni0 interface
+  string ipAddress = 6;
+
+  // Node name on which the application is running
+  string nodeName = 7;
+}
+
+message KubeVolumeReplicaInfo {
+  // Name of the volume replica
+  string name = 1;
+
+  // Rebuild progress of the volume replica
+  uint32 rebuildProgressPercentage = 2;
+
+  // Replica status
+  StorageVolumeReplicaStatus status = 3;
+}
+
+message KubeVolumeInfo {
+  // Name of the volume
+  string name = 1;
+
+  // Status of the volume
+  StorageVolumeState state = 2;
+
+  // Robustness of the volume
+  StorageVolumeRobustness robustness = 3;
+
+  // Age of the volume, seconds ago
+  uint64 age = 4;
+
+  // Provisioned size of the volume in bytes
+  uint64 provisionedBytes = 5;
+
+  // Allocated size of the volume in bytes
+  uint64 allocatedBytes = 6;
+
+  // PV/PVC status of the volume
+  StorageVolumePVCStatus pvcStatus = 7;
+
+  // Replicas of the volume
+  repeated KubeVolumeReplicaInfo replica = 8;
+}
+
+message KubeStorageInfo {
+  // Overall status of Longhorn
+  StorageHealthStatus health = 1;
+
+  // time of the health status transition, seconds ago
+  uint64 transitionTime = 2;
+
+  // Status of all the volumes in Longhorn
+  repeated KubeVolumeInfo volumes = 3;
+}
+
+enum KubeCompUpgradeStatus {
+  KUBE_COMP_UPGRADE_STATUS_UNKNOWN = 0;
+  KUBE_COMP_UPGRADE_STATUS_DOWNLOAD = 1;
+  KUBE_COMP_UPGRADE_STATUS_DOWNLOAD_FAILED = 2;
+  KUBE_COMP_UPGRADE_STATUS_IN_PROGRESS = 3; 
+  KUBE_COMP_UPGRADE_STATUS_FAILED = 4;
+  KUBE_COMP_UPGRADE_STATUS_COMPLETED = 5;
+}
+
+enum KubeComp {
+  COMP_UNKNOWN = 0;
+  COMP_CONTAINERD = 1; //every node will publish
+  COMP_K3S = 2; // every node will publish
+  COMP_MULTUS = 3; // only the seed node will publish the remaining here and below
+  COMP_KUBEVIRT = 4;
+  COMP_CDI = 5;
+  COMP_LONGHORN = 6;
+} 
+
+message KubeClusterUpgradeStatus {
+  // currentNode will have an empty value when no node is in an upgrade
+  string currentNode = 1;	
+
+  // component currently under upgrade, COMP_UNKNOWN when no upgrades in progress
+  KubeComp component = 2;
+
+  // status of the current component under upgrade, KUBE_COMP_UPGRADE_STATUS_UNKNOWN when 
+  //   no upgrades in progress
+  KubeCompUpgradeStatus status = 3;
+ 
+  // error info in case of failure
+  ErrorInfo error = 4;
+}
+
+message KubeClusterInfo {
```

Comment:
Is this published by every node in the cluster? (I assume it would be a duplicate info)

Reply:
This is published by one node chosen through leader election.